routing/payment_session.go

```diff
@@ -161,11 +175,11 @@ func (p *paymentSession) RequestRoute(payment *LightningPayment,
 			bandwidthHints:  p.bandwidthHints,
 		},
 		&RestrictParams{
-			IgnoredNodes:      pruneView.vertexes,
-			IgnoredEdges:      pruneView.edges,
-			FeeLimit:          payment.FeeLimit,
-			OutgoingChannelID: payment.OutgoingChannelID,
-			CltvLimit:         cltvLimit,
+			ProbabilitySource:     p.getEdgeProbability,
+			FeeLimit:              payment.FeeLimit,
+			OutgoingChannelID:     payment.OutgoingChannelID,
+			CltvLimit:             cltvLimit,
+			PaymentAttemptPenalty: DefaultPaymentAttemptPenalty,
```

Comment:
Nice! This can let us dynamically increase this each time a route is requested (if we want).

Reply:
Yes, it will be interesting to see how this will develop further.